app/shop_trip_moduls/custom_functions.py

```diff
@@ -0,0 +1,133 @@
+from app.shop_trip_moduls.Cars import Car
+from app.shop_trip_moduls.Customers import Customer
+from app.shop_trip_moduls.Locations import Point
+from app.shop_trip_moduls.Shops import Shop
+
+
+def create_data_shops(data: dict) -> list:
+    return [
+        Shop(
+            shop["name"],
+            Point(shop["location"][0], shop["location"][1]),
+            shop["products"]
+        ) for shop in data.get("shops")
+    ]
+
+
+def create_data_customers(data: dict) -> list:
+    return [
+        Customer(
+            customer["name"],
+            customer["product_cart"],
+            Point(customer["location"][0], customer["location"][1]),
+            customer["money"],
+            Car(
+                customer["car"]["brand"],
+                customer["car"]["fuel_consumption"],
+                data.get("FUEL_PRICE")
+            )
+        ) for customer in data.get("customers")
+    ]
+
+
+def play_shop_trip(
+        shops: list,
+        customers: list
+) -> None:
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        purchase_price = 0
+        best_store = ""
+        list_price = []
+        for shop in shops:
+            distance = calculate_distance(
+                customer.location,
+                shop.get_location()
+            )
+            price_trip = calculate_price_trip(
+                customer.car,
+                distance
+            )
+            full_price = count_full_price(
+                price_trip,
+                shop.price_list,
+                customer.product_cart
+            )
+            print(f"{customer.name}'s trip to "
+                  f"the {shop.name} costs {full_price}")
+            list_price.append(full_price)
+        the_cheapest_option = min(list_price)
+        for num in range(len(list_price)):
+            if list_price[num] == the_cheapest_option:
+                best_store = shops[num]
+        customer.money = customer.money - the_cheapest_option
+        if customer.money < 0:
+            print(f"{customer.name} doesn't have enough money "
+                  f"to make a purchase in any shop")
+            continue
+        print(f"{customer.name} rides to {best_store.name}\n")
+        print("Date: 04/01/2021 12:33:41")

```

Comment:
Why does datetime hardcode ?

Reply:
The date written in the test does not correspond to the current one, it is a constant.
That's why I used such a date to pass the tests

Reply:
and `import datetime`, not `from datetime import datetime`